File: app/src/main/java/com/jajinba/pixabaydemo/view/fragment/ImageGridFragment.java
package com.jajinba.pixabaydemo.view.fragment;


import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import com.jajinba.pixabaydemo.R;
import com.jajinba.pixabaydemo.adapter.ImageListAdapter;

public class ImageGridFragment extends ListFragment {

  private static final int SPAN_COUNT = 2;

  private ImageListAdapter mAdapter;

  public static ImageGridFragment newInstance() {
    return new ImageGridFragment();
  }

  @Override
  protected ImageListAdapter getAdapter() {
    if (mAdapter == null) {
      mAdapter = new ImageListAdapter(ImageGridFragment.this);
    }

    return mAdapter;
  }

  @Override
  protected RecyclerView.LayoutManager getLayoutManager() {
    return new StaggeredGridLayoutManager(SPAN_COUNT, StaggeredGridLayoutManager.VERTICAL);
  }

  @Override
  protected int getContentLayout() {
    return R.layout.fragment_list;
  }

}

File: app/src/main/java/com/jajinba/pixabaydemo/view/fragment/ImageGridFragment.kt
package com.jajinba.pixabaydemo.view.fragment

import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import com.jajinba.pixabaydemo.model.ImagesSearchDataSource
import com.jajinba.pixabaydemo.model.PixabayImageObject

class ImageGridFragment(imageSearchDataSource: ImagesSearchDataSource) : ListFragment(imageSearchDataSource) {

  companion object {
    private const val SPAN_COUNT = 2
    fun newInstance(imageSearchDataSource: ImagesSearchDataSource): ImageGridFragment {
      return ImageGridFragment(imageSearchDataSource)
    }
  }

  override fun getLayoutManager(): RecyclerView.LayoutManager {
    return StaggeredGridLayoutManager(SPAN_COUNT, StaggeredGridLayoutManager.VERTICAL)
  }

  override fun setImageList(keyword: String, imageList: MutableList<PixabayImageObject>) {
    searchFinished(keyword, imageList)
  }

  override fun onFailed(errorMsg: Int) {
    // TODO
  }

}
